x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -1254,34 +1260,95 @@ public static Map<String, Object> runEsql(
         AssertWarnings assertWarnings,
         Mode mode,
         boolean checkPartialResults
+    ) throws IOException {
+        return runEsql(requestObject, assertWarnings, null, mode, checkPartialResults);
+    }
+
+    public static Map<String, Object> runEsql(
+        RequestObjectBuilder requestObject,
+        AssertWarnings assertWarnings,
+        @Nullable ProfileLogger profileLogger,
+        Mode mode,
+        boolean checkPartialResults
     ) throws IOException {
         var results = mode == ASYNC
-            ? runEsqlAsync(requestObject, randomBoolean(), assertWarnings)
-            : runEsqlSync(requestObject, assertWarnings);
-        return checkPartialResults ? assertNotPartial(results) : results;
+            ? runEsqlAsync(requestObject, randomBoolean(), assertWarnings, profileLogger)
+            : runEsqlSync(requestObject, assertWarnings, profileLogger);
+        if (checkPartialResults) {
+            assertNotPartial(results);
+        }
+        return results;
     }
 
     public static Map<String, Object> runEsql(RequestObjectBuilder requestObject, AssertWarnings assertWarnings, Mode mode)
         throws IOException {
         return runEsql(requestObject, assertWarnings, mode, true);
     }
 
+    public static Map<String, Object> runEsql(RequestObjectBuilder requestObject, AssertWarnings assertWarnings,
```

Comment:
I kept here the overloads without the ProfileLogger param, as they will be easier to track, instead of looking for a null as a param